Static assert failure when compiling procyon as part of antares

clang++ -mmacosx-version-min=10.7 -MMD -MF obj/ext/procyon/src/cpp/src/libprocyon-cpp.value.o.d -DNDEBUG -I../../../ext/procyon/src/cpp/include -I../../../ext/procyon/src/c/include -Os -Wall -Wextra -Wsign-compare -Werror -Wno-initializer-overrides -std=c++11 -stdlib=libc++ -c ../../../ext/procyon/src/cpp/src/value.cpp -o obj/ext/procyon/src/cpp/src/libprocyon-cpp.value.o
../../../ext/procyon/src/cpp/src/value.cpp:158:1: error: static_assert failed "allows char conversion"
static_assert(conversion<value, char>::fails, "allows char conversion");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../ext/procyon/src/cpp/src/value.cpp:159:1: error: static_assert failed "allows signed char conversion"
static_assert(conversion<value, signed char>::fails, "allows signed char conversion");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../ext/procyon/src/cpp/src/value.cpp:160:1: error: static_assert failed "allows unsigned char conversion"
static_assert(conversion<value, unsigned char>::fails, "allows unsigned char conversion");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../ext/procyon/src/cpp/src/value.cpp:163:1: error: static_assert failed "allows void* conversion"
static_assert(conversion<value, void*>::fails, "allows void* conversion");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../ext/procyon/src/cpp/src/value.cpp:164:1: error: static_assert failed "allows char* conversion"
static_assert(conversion<value, char*>::fails, "allows char* conversion");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
5 errors generated.

--- a/src/cpp/src/value.cpp
+++ b/src/cpp/src/value.cpp
@@ -155,6 +155,8 @@ static_assert(conversion<value, const string&>::fails, "allows string copy conve
 static_assert(conversion<value, const array&>::fails, "allows array copy conversion");
 static_assert(conversion<value, const map&>::fails, "allows map copy conversion");
 
+// Broken for some reason prior to Xcode 7.3.
+#if defined(__apple_build_version__) && (__apple_build_version__ < 7030000)
 static_assert(conversion<value, char>::fails, "allows char conversion");
 static_assert(conversion<value, signed char>::fails, "allows signed char conversion");
 static_assert(conversion<value, unsigned char>::fails, "allows unsigned char conversion");
@@ -162,5 +164,6 @@ static_assert(conversion<value, uint32_t>::fails, "allows uint32_t conversion");
 static_assert(conversion<value, uint64_t>::fails, "allows uint64_t conversion");
 static_assert(conversion<value, void*>::fails, "allows void* conversion");
 static_assert(conversion<value, char*>::fails, "allows char* conversion");
+#endif
 
 }  // namespace pn

